loopchain/peer/peer_outer_service.py

```diff
@@ -235,6 +235,9 @@ def GetStatus(self, request, context):
         if request.request == 'block_sync':
             try:
                 status_data = channel_stub.sync_task().get_status()
+                if (status_data['state'] == 'BlockGenerate') and (status_data['block_height'] > 1):
+                    status_data = None
+                    raise RuntimeError("Cannot accept BlockSync during BlockGenerate state")
```

Comment:
Can you explain why did you set `status_data` as `None`, which raises `ChannelStatusError`?

Reply:
That's because if status_data is not None, it (unnecessarily) caches the status_data. The purpose of this feature is to ignore block_sync request when the node gets leader. Because `StatusReply` doesn't have response_code as parameter, I couldn't send back to the requester with specific error code so let `ChannelStatusError` to be raised.

Reply:
Changed again. It won't raise `ChannelStatusError` from Leader node. Instead, when the requester node send GetStatus(block_sync), it would recognize from status_data whether it's leader or not.